server/config/config.go

```diff
@@ -1008,8 +1007,21 @@ func (c *Config) RewriteFile(new *Config) error {
 	if err := toml.NewEncoder(&buf).Encode(*new); err != nil {
 		return err
 	}
-	// 0644 represent the permission of file which is "-rw-r--r--" here.
-	return ioutil.WriteFile(filePath, buf.Bytes(), 0644)
+	dir := filepath.Dir(filePath)
+	tmpfile := filepath.Join(dir, "tmp_pd.toml")
+
+	f, err := os.Create(tmpfile)
```

Comment:
The default permission is 0666, should we change it to 0644?

Reply:
I think it's unnecessary?